src/v/pandaproxy/server.h

```diff
@@ -161,6 +180,54 @@ class auth_ctx_server : public ctx_server<service_t> {
         // The service
         service_t& service() const { return context().service; };
 
+        template<std::invocable<kafka_client_cache&> Func>
+        auto dispatch(Func&& func) {
+            return service().client_cache().invoke_on_cache(
+              user, std::forward<Func>(func));
+        }
+
+        template<std::invocable<kafka::client::client&> Func>
+        auto dispatch(Func&& func) {
+            switch (authn_method) {
+            case config::rest_authn_method::none: {
+                return std::forward<Func>(func)(service().client().local());
+            }
+            case config::rest_authn_method::http_basic: {
+                return dispatch([this, &func](kafka_client_cache& cache) {
```

Comment:
i thought you had to also forward into the capture? not sure tho

Reply:
Once it;s named, it's an lvalue, so that doesn't really work (`[thing{std::forward<decltype(thing)>(thing)}]`}. I get away with it here because I don't ever need a `const&`.